identity_iota_core/src/document/iota_document.rs

```diff
@@ -293,6 +299,59 @@ impl IotaDocument {
     }
     StateMetadataDocument::unpack(state_metadata).and_then(|doc| doc.into_iota_document(did))
   }
+
+  /// Deserializes the document from an Alias Output.
+  ///
+  /// If `allow_empty` is true, this will return an empty DID document marked as `deactivated`
+  /// if `state_metadata` is empty.
+  ///
+  /// NOTE: `did` is required since it is omitted from the serialized DID Document and
+  /// cannot be inferred from the state metadata. It also indicates the network, which is not
+  /// encoded in the `AliasId` alone.
+  pub fn unpack_from_output(
+    did: &IotaDID,
+    alias_output: &AliasOutput,
+    allow_empty: bool,
+    network_name: &NetworkName,
+  ) -> Result<IotaDocument> {
+    if alias_output.state_metadata().is_empty() && allow_empty {
+      let mut empty_document = IotaDocument::new_with_id(did.clone());
+      empty_document.metadata.created = None;
+      empty_document.metadata.updated = None;
+      empty_document.metadata.deactivated = Some(true);
+      return Ok(empty_document);
+    }
+    let mut iota_document: IotaDocument =
+      StateMetadataDocument::unpack(alias_output.state_metadata()).and_then(|doc| doc.into_iota_document(did))?;
+    iota_document.set_controller_and_governor_addresses(alias_output, network_name);
```

Comment:
Don't duplicate `unpack` logic here.

Otherwise we should remove `unpack` entirely if it can cause problems by bypassing the setting of controller and governor addresses?